eng/templates/jobs/source-build.yml

```diff
@@ -1,139 +1,103 @@
+      ${{ property.key }}: ${{ property.value }}
+
+    ${{ if ne(platform.container, '') }}:
+      container: ${{ platform.container }}
+    ${{ if ne(platform.pool, '') }}:
+      pool: ${{ platform.pool }}
+
+    workspace:
+      clean: all
+
+    variables:
+    - name: _BuildConfig
+      value: Release
+    - name: _InternalBuildArgs
+      value: ''
+    - name: _TargetRidArgs
+      value: ''
+    - ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
       - name: _InternalBuildArgs
-        value: ''
-      - name: _NonPortableBuildArgs
-        value: ''
-      - ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
-        - name: _InternalBuildArgs
-          value: >-
-            /p:DotNetPublishUsingPipelines=true
-            /p:OfficialBuildId=$(BUILD.BUILDNUMBER)
-      - ${{ if ne(platform.targetRID, '') }}:
-        - name: _NonPortableBuildArgs
-          value: >-
-            /p:ArcadeBuildNonPortable=true
-            /p:TargetRid=${{ platform.targetRID }}
-
-      steps:
-      - script: |
-          set -x
-          df -h
-          ${{ parameters.buildScript }} --ci \
-            --configuration $(_BuildConfig) \
-            --restore --build --pack --publish \
-            $(_InternalBuildArgs) \
-            $(_NonPortableBuildArgs) \
-           /p:ArcadeBuildFromSource=true
-        displayName: Build
-
-      # Upload build logs for diagnosis.
-      - task: CopyFiles@2
-        displayName: Prepare BuildLogs staging directory
-        inputs:
-          SourceFolder: '$(Build.SourcesDirectory)'
-          Contents: |
-            **/*.log
-            **/*.binlog
-            artifacts/source-build/self/prebuilt-report/**
-          TargetFolder: '$(Build.StagingDirectory)/BuildLogs'
-          CleanTargetFolder: true
-        continueOnError: true
-        condition: succeededOrFailed()
-
-      - task: PublishPipelineArtifact@1
-        displayName: Publish BuildLogs
-        inputs:
-          targetPath: '$(Build.StagingDirectory)/BuildLogs'
-          artifactName: BuildLogs
-        continueOnError: true
-        condition: succeededOrFailed()
+        value: >-
+          /p:DotNetPublishUsingPipelines=true
+          /p:OfficialBuildId=$(BUILD.BUILDNUMBER)
+    - ${{ if ne(platform.targetRID, '') }}:
+      - name: _TargetRidArgs
+        value: /p:TargetRid=${{ platform.targetRID }}
+
+    steps:
+    - script: |
+        set -x
+        df -h
+        ${{ parameters.buildScript }} --ci \
+          --configuration $(_BuildConfig) \
+          --restore --build --pack --publish \
+          $(_InternalBuildArgs) \
+          $(_TargetRidArgs) \
+          /p:SourceBuildNonPortable=${{ platform.nonPortable }} \
+          /p:ArcadeBuildFromSource=true
+      displayName: Build
+
+    # Upload build logs for diagnosis.
+    - task: CopyFiles@2
```

Comment:
Not necessarily specific to this PR, but do you think we want to upload the intermediate nupkg for non-official builds?  It seems like it might be helpful for debugging issues.

Reply:
It would, but the repos don't even do that with their own artifacts today, so I expect they won't be ok adding this for source-build. (I've seen some that didn't even save binlogs, which is somewhat terrifying and I won't go that far.) I haven't heard of any particular storage constraints but it would be a *lot* of data (GB per build, depending on repo--on every single PR).

With arcade-powered source-build I'm hoping that it's quickly reproducible enough that people can follow similar approaches for debugging bad artifacts as they do with the existing lack of package artifacts saved in PR validation.